projects/benqi-lending/functions/repayBorrow.spec.ts

```diff
@@ -0,0 +1,168 @@
+            functionName: 'underlying',
+            args: [],
+        });
+
+        expect(checkToApprove).toHaveBeenCalledWith(
+            expect.objectContaining({
+                args: {
+                    account: props.account,
+                    target: underlyingAssetAddress,
+                    spender: CORE_MARKETS.USDC,
+                    amount: parseUnits(props.amount, MARKET_DECIMALS),
+                },
+            }),
+        );
+
+        expect(tools.sendTransactions).toHaveBeenCalledWith(
+            expect.objectContaining({
+                transactions: [
+                    {
+                        target: CORE_MARKETS.USDC,
+                        data: encodeFunctionData({
+                            abi: qiERC20Abi,
+                            functionName: 'repayBorrow',
+                            args: [parseUnits(props.amount, MARKET_DECIMALS)],
+                        }),
+                    },
+                ],
+            }),
+        );
+        expect(result.data).toMatch(`Successfully repaid borrow of ${props.amount} tokens.`);
+    });
+
+    it('should deposit collateral in AVAX tokens on core market', async () => {
+        const props: Parameters<typeof repayBorrow>[0] = {
+            account: '0x0000000000000000000000000000000000000000',
+            chainName: 'Avalanche',
+            amount: '2.12',
+            marketType: 'core',
+            marketName: 'AVAX',
+        };
+
+        const tools: Parameters<typeof repayBorrow>[1] = {
+            sendTransactions: vi.fn().mockReturnValue(Promise.resolve({ data: ['Result'], isMultisig: false })),
+            notify: vi.fn(),
+            getProvider: vi.fn(),
+        };
+
+        const result = await repayBorrow(props, tools);
+
+        expect(checkToApprove).not.toHaveBeenCalled();
+
+        expect(tools.sendTransactions).toHaveBeenCalledWith(
+            expect.objectContaining({
+                transactions: [
+                    {
+                        target: CORE_MARKETS.AVAX,
+                        data: encodeFunctionData({
+                            abi: qiAvaxAbi,
+                            functionName: 'repayBorrow',
+                            args: [],
+                        }),
+                        value: parseUnits(props.amount, MARKET_DECIMALS),
+                    },
+                ],
+            }),
+        );
+        expect(result.data).toMatch(`Successfully repaid borrow of ${props.amount} tokens.`);
+    });
+
+    it('should deposit collateral in ERC20 tokens on ecosystem market', async () => {
+        const props: Parameters<typeof repayBorrow>[0] = {
+            account: '0x0000000000000000000000000000000000000000',
+            chainName: 'Avalanche',
+            amount: '2.12',
+            marketType: 'ecosystem',
+            marketName: 'USDC',
+        };
+
+        const underlyingAssetAddress = '0x1234567890abcdef1234567890abcdef12345678';
```

Comment:
For tidiness I would recommend storing all used addresses in tests as constants and just recycling them

Reply:
IMO both have some maintenance cost associated with them - your approach requires us to create a finite list of mocked addresses and import them in test files. With such limited amount of functions here I don't really see such big benefit